lib/ZwaveClient.ts

```diff
@@ -961,6 +980,173 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
 		}
 	}
 
+	/**
+	 * If the node supports Schedule Lock CC parses all available schedules and cache them
+	 */
+	async getSchedules(nodeId: number) {
+		const zwaveNode = this.getNode(nodeId)
+
+		if (!zwaveNode?.commandClasses['Schedule Entry Lock'].isSupported()) {
+			throw new Error(
+				'Schedule Entry Lock CC not supported on node ' + nodeId
+			)
+		}
+
+		const userCodes = await zwaveNode.commandClasses[
+			'User Code'
+		].getUsersCount()
```

Comment:
This will request the number of users from the lock again. This information **should** be queried during the interview, so you can read the cached information using
```ts
UserCodeCC.getSupportedUsersCached(driver, zwaveNode)
```

https://github.com/zwave-js/node-zwave-js/blob/4854b0cf2b2c9bad15ac1b42a2632776e28db4b7/packages/cc/src/cc/UserCodeCC.ts#L1007-L1015